src/Symfony/Component/Validator/Constraints/PasswordStrengthValidator.php

```diff
@@ -15,15 +15,26 @@
 use Symfony\Component\Validator\ConstraintValidator;
 use Symfony\Component\Validator\Exception\UnexpectedTypeException;
 use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Symfony\Component\Validator\Password\PasswordStrengthEstimatorInterface;
 
 final class PasswordStrengthValidator extends ConstraintValidator
 {
-    /**
-     * @param (\Closure(string):PasswordStrength::STRENGTH_*)|null $passwordStrengthEstimator
-     */
+    private \Closure $passwordStrengthEstimator;
+
     public function __construct(
-        private readonly ?\Closure $passwordStrengthEstimator = null,
+        \Closure|PasswordStrengthEstimatorInterface $passwordStrengthEstimator,
     ) {
+        if ($passwordStrengthEstimator instanceof \Closure) {
+            // The target for 8.0 is to remove the support for Closure and only allow the Interface
+            trigger_deprecation('symfony/validator', '7.2', 'Passing a closure to "%s()" is deprecated, pass an instance of "%s" instead.', __METHOD__, PasswordStrengthEstimatorInterface::class);
+        }
+
+        // Still needed in this version to support the closure
+        if ($passwordStrengthEstimator instanceof \Closure) {
+            $this->passwordStrengthEstimator = $passwordStrengthEstimator;
+        } else {
+            $this->passwordStrengthEstimator = $passwordStrengthEstimator->estimateStrength(...);
+        }
```

Comment:
I would rather wrap the closure into an anonymous class implementing the new interface. That gives us less code to change in 8.0.

Reply:
updated